azureml_tutorial/train_with_mlflow.py

```diff
@@ -23,6 +23,8 @@ def main() -> None:
 
     args = parse_args()
 
+    # This can be used to log all metrics and artifacts that are generated by the model.
+    mlflow.autolog(log_models=True)
```

Comment:
I'm not a big fan of this function since it feels a bit magical.
I prefer to keep the code in such a way that I can directly see what happens.
The main reason for this is that the target audience for this tutorial is people with limited to no experience with AzureML or MLflow.
So, if you really want to add this function, you will need to explain it in the tutorial.